IdentityCore/src/cache/token/MSIDCommonCredential.m

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import <Foundation/Foundation.h>
+#import "MSIDClientInfo.h"
+#import "MSIDCommonCredential.h"
+#import "MSIDCommonLogger.h"
+#import "MSIDCommonStorageConstants.h"
+#import "MSIDCredentialType.h"
+#import "NSDate+MSIDExtensions.h"
+
+@interface MSIDCommonCredential ()
+
+@property (readwrite) NSDictionary *json;
+
+@end
+
+@implementation MSIDCommonCredential
+
+- (nullable instancetype)init {
+    self = [super init];
+    if (self) {
+        // defaults...
+    }
+
+    return self;
+}
+
+- (nullable instancetype)initWithType:(MSIDCredentialType)credentialType
+                        homeAccountId:(nonnull NSString *)homeAccountId
+                          environment:(nonnull NSString *)environment
+                             clientId:(nullable NSString *)clientId
+                               secret:(nullable NSString *)secret
+                             familyId:(nullable NSString *)familyId
+                               target:(nullable NSString *)target
+                           clientInfo:(nullable MSIDClientInfo *)clientInfo
+                                realm:(nullable NSString *)realm
+                     additionalFields:(nullable NSDictionary *)additionalFields {
+    MSID_TRACE;
+    self.credentialType = credentialType;
```

Comment:
nit: Apple recommends to avoid using "self" in the initializer and use ivars instead.

Reply:
You're right, I'll update it and keep this in mind going forward.